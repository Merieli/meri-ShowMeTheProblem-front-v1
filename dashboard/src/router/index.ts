import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router';

const Home = () => import('../views/Home/PageHome.vue');
const Feedbacks = () => import('../views/Feedbacks/PageFeedbacks.vue');
const Credentials = () => import('../views/Credentials/PageCredentials.vue');

export const routes: Array<RouteRecordRaw> = [
    {
        path: '/',
        name: 'home',
        component: Home,
    },
    {
        path: '/feedbacks',
        name: 'Feedbacks',
        component: Feedbacks,
        meta: {
            hasAuth: true,
        },
    },
    {
        path: '/Credentials',
        name: 'Credentials',
        component: Credentials,
        meta: {
            hasAuth: true,
        },
    },
    {
        path: '/:pathMatch(.*)*',
        redirect: { name: 'Home' },
    },
];

const router = createRouter({
    history: createWebHistory('/'),
    routes,
});

export default router;
