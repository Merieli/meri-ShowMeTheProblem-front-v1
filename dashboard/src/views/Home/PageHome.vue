<template>
    <div class="home">
        <GeralHeader />
        <section class="home__presentation">
            <div class="home__content flex flex-col mt-28">
                <h2 class="home__title home__title_light">
                    Tenha um feedback.
                    <br class="hidden lg:inline-block" />
                    E faça seus clientes mais felizes!
                </h2>
                <p class="home__text home__text_light">
                    Receba ideias, reclamações e feedbacks com um simples widget na página.
                </p>
                <BaseButton id="create-account-button" color="light" @click="toggleModalCreate">
                    Crie uma conta grátis
                </BaseButton>
            </div>
        </section>
        <section class="home__contact">
            <h3 class="home__title home__title_dark">Alguma dúvida?</h3>
            <p class="home__text">Quer saber melhor como funciona e quais são os preços?</p>
            <BaseButton color="dark">
                <a href="mailto:merieli.dev@gmail.com">Nos mande um e-mail</a>
            </BaseButton>
        </section>
        <footer class="home__footer">
            <p class="home__text">ShowMeTheProblem © 2022</p>
        </footer>
        <PartModal type-modal="create" :open="openCreateModal" @close="toggleModalCreate" />
    </div>
</template>

<script lang="ts">
import BaseButton from '../../components/BaseButton/index.vue';
import GeralHeader from '../../components/GeralHeader.vue';
import PartModal from '../../components/PartModal.vue';
import { defineComponent, ref } from 'vue';

export default defineComponent({
    name: 'PageHome',
    components: {
        GeralHeader,
        BaseButton,
        PartModal,
    },
    setup() {
        const openCreateModal = ref(false);

        const toggleModalCreate = () => {
            openCreateModal.value = !openCreateModal.value;
        };

        return {
            openCreateModal,
            toggleModalCreate,
        };
    },
});
</script>

<style scoped lang="scss">
.home {
    display: flex;
    flex-direction: column;
    align-items: center;

    &__title {
        font-size: theme('fontSize.4xl');
        font-family: theme('fontFamily.black');
        line-height: 1.1;

        &_light {
            color: theme('colors.brand.white');
        }

        &_dark {
            color: theme('colors.brand.black');
        }
    }

    &__text {
        color: theme('colors.brand.black');
        font-size: theme('fontSize.xl');
        font-family: theme('fontFamily.regular');
        padding-bottom: 2.5rem;

        &_light {
            color: theme('colors.brand.white');
        }

        &_dark {
            color: theme('colors.brand.black');
        }
    }

    &__presentation {
        background-color: theme('colors.brand.main');
        display: flex;
        justify-content: center;
        width: 100%;
    }

    &__content {
        padding: 1rem;
        width: 100%;
        min-height: 500px;
        max-width: 1080px;
        background-size: contain;
        background-position: right 10% bottom 0%;
        background-image: url('../../assets/images/blue-balloons.png');
        background-repeat: no-repeat;
    }

    &__contact {
        text-align: center;
        padding: 4rem;
        max-width: 1080px;
    }

    &__footer {
        text-align: center;
        padding: 2.5rem;
        background-color: theme('colors.brand.gray');
        width: 100%;

        .home__text {
            padding: 0;
            font-family: theme('fontFamily.bold');
        }
    }
}
</style>
