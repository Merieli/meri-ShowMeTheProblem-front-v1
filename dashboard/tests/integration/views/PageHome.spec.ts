import { routes } from '@/router'
import PageHome from '@/views/Home/PageHome.vue'
import { shallowMount, VueWrapper } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'

describe('PageHome.vue', () => {
    const router = createRouter({
        history: createWebHistory(),
        routes,
    })

    let wrapper: VueWrapper

    beforeEach(async () => {
        router.push('./')
        await router.isReady()

        wrapper = shallowMount(PageHome, {
            global: {
                plugins: [router],
            },
        })
    })
    describe('Integração/Componente', () => {
        describe('👀 Renderização:', () => {
            test('Dado o componente Quando renderizado Então deve possuir html equivalente ao Snapshot gravado', async () => {
                expect(wrapper.html()).toMatchSnapshot()
            })
        })
        describe('🧠 Comportamento:', () => {
            return
        })
        describe('🐕 Navegação:', () => {
            return
        })
    })
})
