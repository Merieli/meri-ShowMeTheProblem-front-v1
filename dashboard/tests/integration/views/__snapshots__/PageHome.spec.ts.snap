// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHome.vue Integra√ß√£o/Componente üëÄ Renderiza√ß√£o: Dado o componente Quando renderizado Ent√£o deve possuir html equivalente ao Snapshot gravado 1`] = `
"<div class=\\"home\\">
  <geral-header-stub></geral-header-stub>
  <section class=\\"home__presentation\\">
    <div class=\\"home__content flex flex-col mt-28\\">
      <h2 class=\\"home__title home__title_light\\"> Tenha um feedback. <br class=\\"hidden lg:inline-block\\"> E fa√ßa seus clientes mais felizes! </h2>
      <p class=\\"home__text home__text_light\\"> Receba ideias, reclama√ß√µes e feedbacks com um simples widget na p√°gina. </p>
      <base-button-stub color=\\"light\\" typebutton=\\"button\\" isdisabled=\\"false\\"></base-button-stub>
    </div>
  </section>
  <section class=\\"home__contact\\">
    <h3 class=\\"home__title home__title_dark\\">Alguma d√∫vida?</h3>
    <p class=\\"home__text\\">Quer saber melhor como funciona e quais s√£o os pre√ßos?</p>
    <base-button-stub color=\\"dark\\" typebutton=\\"button\\" isdisabled=\\"false\\"></base-button-stub>
  </section>
  <footer class=\\"home__footer\\">
    <p class=\\"home__text\\">ShowMeTheProblem ¬© 2022</p>
  </footer>
  <part-modal-stub typemodal=\\"create\\" open=\\"false\\"></part-modal-stub>
</div>"
`;
